### ISSUE TITLE
SIGSEGV Sometimes When Closing a Program by Clicking a Button

### ISSUE THREAD

**Issue by justdan96** [fyne-io/fyne#1604]
### Describe the bug:
I've noticed that sometimes my program faults with "[signal SIGSEGV: segmentation violation code=0x1 addr=0xf8 pc=0x7fdae2aa6fc4]". It doesn't happen every time, though, so I'm not sure exactly where it is going wrong, full error is here: https://hastebin.com/efiyuxofag.go

### To Reproduce:
Steps to reproduce the behaviour:
* Run any Fyne program on my system which closes the window by clicking a button widget, it will sometimes need to be ran lots of times until I can reproduce the exact error message.

### Example Code

```
package main

import (
	"fyne.io/fyne/app"
	"fyne.io/fyne/container"
	"fyne.io/fyne/layout"
	"fyne.io/fyne/widget"
	"fmt"
)


func showConfirm(title, question string, callback func(bool)) {
	a := app.New()
	w := a.NewWindow(title)
	yes := widget.NewButton("Yes", func() {
		callback(true)
		w.Close()
	})
	yes.Importance = widget.HighImportance
	w.SetContent(container.NewVBox(
		widget.NewLabel(question),
		container.NewHBox(layout.NewSpacer(),
			widget.NewButton("No", func() {
				callback(false)
				w.Close()
			}), yes,
			layout.NewSpacer())))
	w.ShowAndRun()
}

func main() {
	showConfirm("MyTitle", "Are you sure?", func(b bool) {
		if b {
			fmt.Println("Confirmed")
		}
	})
}
```

### Device (please complete the following information):
 - OS: Ubuntu Linux
 - Version: 20.04
 - Go version: 1.12.1
 - Fyne version: 0ca05dc12c03b8873d2f22caa6ce4c28988b3886

**Comment by justdan96** [comment]
```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xf8 pc=0x7fdae2aa6fc4]

runtime stack:
runtime.throw(0x9b3f79, 0x2a)
	/usr/local/go/src/runtime/panic.go:617 +0x72
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:374 +0x4a9

goroutine 24 [syscall, locked to thread]:
runtime.cgocall(0x868240, 0xc0000d7d50, 0x1c94400)
	/usr/local/go/src/runtime/cgocall.go:128 +0x5b fp=0xc0000d7d20 sp=0xc0000d7ce8 pc=0x428e2b
fyne.io/fyne/vendor/github.com/go-gl/gl/v3.2-core/gl._Cfunc_glowClear(0x7fdae2833700, 0x3e4243c500004100)
	_cgo_gotypes.go:3659 +0x41 fp=0xc0000d7d50 sp=0xc0000d7d20 pc=0x7a8a91
fyne.io/fyne/vendor/github.com/go-gl/gl/v3.2-core/gl.Clear(...)
	/home/dan/go/src/fyne.io/fyne/vendor/github.com/go-gl/gl/v3.2-core/gl/package.go:8667
fyne.io/fyne/internal/painter/gl.(*glPainter).glClearBuffer(0xc000488000)
	/home/dan/go/src/fyne.io/fyne/internal/painter/gl/gl_core.go:188 +0xe5 fp=0xc0000d7d78 sp=0xc0000d7d50 pc=0x7cd055
fyne.io/fyne/internal/painter/gl.(*glPainter).Clear(0xc000488000)
	/home/dan/go/src/fyne.io/fyne/internal/painter/gl/painter.go:51 +0x2b fp=0xc0000d7d90 sp=0xc0000d7d78 pc=0x7cd81b
fyne.io/fyne/internal/driver/glfw.(*glCanvas).paint(0xc000180000, 0xf7, 0x4e)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/canvas.go:412 +0x6b fp=0xc0000d7de8 sp=0xc0000d7d90 pc=0x7f025b
fyne.io/fyne/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/loop.go:164 +0x83 fp=0xc0000d7e38 sp=0xc0000d7de8 pc=0x7fc793
fyne.io/fyne/internal/driver/glfw.(*window).RunWithContext(0xc0000c84e0, 0xc0000d7e60)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:1091 +0x38 fp=0xc0000d7e50 sp=0xc0000d7e38 pc=0x7fb168
fyne.io/fyne/internal/driver/glfw.(*gLDriver).repaintWindow(0xc0000b8190, 0xc0000c84e0)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/loop.go:155 +0x60 fp=0xc0000d7e88 sp=0xc0000d7e50 pc=0x7f34b0
fyne.io/fyne/internal/driver/glfw.(*gLDriver).startDrawThread.func1(0xc0000b8190, 0xc000092480, 0xc0000b8230)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/loop.go:208 +0x231 fp=0xc0000d7fc8 sp=0xc0000d7e88 pc=0x7fca81
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc0000d7fd0 sp=0xc0000d7fc8 pc=0x47e011
created by fyne.io/fyne/internal/driver/glfw.(*gLDriver).startDrawThread
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/loop.go:182 +0xbd

goroutine 1 [semacquire, locked to thread]:
sync.runtime_Semacquire(0xc0000c8650)
	/usr/local/go/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc0000c8648)
	/usr/local/go/src/sync/waitgroup.go:130 +0x65
fyne.io/fyne/internal/driver/glfw.(*window).waitForEvents(...)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:1158
fyne.io/fyne/internal/driver/glfw.(*window).destroy(0xc0000c84e0, 0xc0000b8190)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:491 +0xa9
fyne.io/fyne/internal/driver/glfw.(*gLDriver).runGL(0xc0000b8190)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/loop.go:121 +0x545
fyne.io/fyne/internal/driver/glfw.(*gLDriver).Run(0xc0000b8190)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/driver.go:71 +0x37
fyne.io/fyne/internal/driver/glfw.(*window).ShowAndRun(0xc0000c84e0)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:436 +0x57
main.showTextDialog(0x9983bd, 0x4, 0x9af4ca, 0x21, 0xa3f540, 0xc00009f350, 0x9c0870)
	/home/dan/Code/genity/fyne-dialogs2/main.go:36 +0x39c
main.showInformation(0x9983bd, 0x4, 0x9af4ca, 0x21, 0x9c0870)
	/home/dan/Code/genity/fyne-dialogs2/main.go:14 +0x80
main.main()
	/home/dan/Code/genity/fyne-dialogs2/main.go:69 +0x57

goroutine 19 [sleep]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:307
time.Sleep(0x22ecb25c00)
	/usr/local/go/src/runtime/time.go:105 +0x159
fyne.io/fyne/internal/painter.svgCacheJanitor.func1()
	/home/dan/go/src/fyne.io/fyne/internal/painter/svg_cache.go:51 +0x109
sync.(*Once).Do(0x10ef390, 0xc0000ae300)
	/usr/local/go/src/sync/once.go:44 +0xb3
created by fyne.io/fyne/internal/painter.svgCacheJanitor
	/home/dan/go/src/fyne.io/fyne/internal/painter/svg_cache.go:49 +0x9b

goroutine 20 [chan receive]:
fyne.io/fyne/app.newAppWithDriver.func1(0xc0000922a0, 0xc00015c000)
	/home/dan/go/src/fyne.io/fyne/app/app.go:126 +0x3d
created by fyne.io/fyne/app.newAppWithDriver
	/home/dan/go/src/fyne.io/fyne/app/app.go:124 +0x1ea

goroutine 21 [syscall]:
syscall.Syscall6(0xe8, 0x5, 0xc0001a1d14, 0x7, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
fyne.io/fyne/vendor/golang.org/x/sys/unix.EpollWait(0x5, 0xc0001a1d14, 0x7, 0x7, 0xffffffffffffffff, 0xc0001a1d10, 0x0, 0xc000086a80)
	/home/dan/go/src/fyne.io/fyne/vendor/golang.org/x/sys/unix/zsyscall_linux_amd64.go:76 +0x72
fyne.io/fyne/vendor/github.com/fsnotify/fsnotify.(*fdPoller).wait(0xc0000c0140, 0x9c0300, 0xc0000c0140, 0x0)
	/home/dan/go/src/fyne.io/fyne/vendor/github.com/fsnotify/fsnotify/inotify_poller.go:86 +0x91
fyne.io/fyne/vendor/github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc0000b81e0)
	/home/dan/go/src/fyne.io/fyne/vendor/github.com/fsnotify/fsnotify/inotify.go:192 +0x18a
created by fyne.io/fyne/vendor/github.com/fsnotify/fsnotify.NewWatcher
	/home/dan/go/src/fyne.io/fyne/vendor/github.com/fsnotify/fsnotify/inotify.go:59 +0x1a8

goroutine 22 [chan receive]:
fyne.io/fyne/app.watchFile.func1(0xc0000b81e0, 0xc0000aa120, 0x24, 0xc00009d2d0, 0xc00009d2e0)
	/home/dan/go/src/fyne.io/fyne/app/settings_desktop.go:42 +0x58
created by fyne.io/fyne/app.watchFile
	/home/dan/go/src/fyne.io/fyne/app/settings_desktop.go:41 +0xe7

goroutine 25 [sleep]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:307
time.Sleep(0xee6b280)
	/usr/local/go/src/runtime/time.go:105 +0x159
fyne.io/fyne/widget.(*Button).Tapped.func1(0xc0001b2000)
	/home/dan/go/src/fyne.io/fyne/widget/button.go:190 +0x2a
fyne.io/fyne/widget.(*Button).Tapped(0xc0001b2000, 0xc0000c00c0)
	/home/dan/go/src/fyne.io/fyne/widget/button.go:199 +0x72
fyne.io/fyne/internal/driver/glfw.(*window).mouseClicked.func5()
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:734 +0x38
fyne.io/fyne/internal/driver/glfw.(*window).runEventQueue(0xc0000c84e0)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:1152 +0x7a
created by fyne.io/fyne/internal/driver/glfw.(*gLDriver).createWindow.func1
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/window.go:1176 +0x103

goroutine 26 [chan receive]:
fyne.io/fyne/internal/driver/glfw.(*glCanvas).setupThemeListener.func1(0xc0000924e0, 0xc000180000)
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/canvas.go:458 +0x47
created by fyne.io/fyne/internal/driver/glfw.(*glCanvas).setupThemeListener
	/home/dan/go/src/fyne.io/fyne/internal/driver/glfw/canvas.go:456 +0xa6

goroutine 5 [chan receive]:
fyne.io/fyne/internal/painter.SvgCacheMonitorTheme.func1(0xc000026060)
	/home/dan/go/src/fyne.io/fyne/internal/painter/svg_cache.go:103 +0x34
created by fyne.io/fyne/internal/painter.SvgCacheMonitorTheme
	/home/dan/go/src/fyne.io/fyne/internal/painter/svg_cache.go:101 +0x9c
```

**Comment by stuartmscott** [comment]
I've seen something similar, tho this was triggered when I clicked the window close button:

```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x0]

runtime stack:
runtime.throw(0x44db9a6, 0x2a)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/usr/local/Cellar/go/1.15/libexec/src/runtime/signal_unix.go:704 +0x48c

goroutine 34 [syscall, locked to thread]:
runtime.cgocall(0x43e8940, 0xc000721e00, 0x437a488)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/cgocall.go:133 +0x5b fp=0xc000721dd0 sp=0xc000721d98 pc=0x4005dbb
fyne.io/fyne/vendor/github.com/go-gl/glfw/v3.3/glfw._Cfunc_glfwMakeContextCurrent(0x0)
	_cgo_gotypes.go:1446 +0x45 fp=0xc000721e00 sp=0xc000721dd0 pc=0x43612c5
fyne.io/fyne/vendor/github.com/go-gl/glfw/v3.3/glfw.DetachCurrentContext()
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/vendor/github.com/go-gl/glfw/v3.3/glfw/context.go:23 +0x2a fp=0xc000721e38 sp=0xc000721e00 pc=0x436346a
fyne.io/fyne/internal/driver/glfw.(*window).RunWithContext(0xc00012a000, 0xc000721e60)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/internal/driver/glfw/window.go:1102 +0x3d fp=0xc000721e50 sp=0xc000721e38 pc=0x4378dfd
fyne.io/fyne/internal/driver/glfw.(*gLDriver).repaintWindow(0xc000094140, 0xc00012a000)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/internal/driver/glfw/loop.go:155 +0x65 fp=0xc000721e88 sp=0xc000721e50 pc=0x436fe05
fyne.io/fyne/internal/driver/glfw.(*gLDriver).startDrawThread.func1(0xc000094140, 0xc000100060, 0xc000120000)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/internal/driver/glfw/loop.go:208 +0x234 fp=0xc000721fc8 sp=0xc000721e88 pc=0x437a794
runtime.goexit()
	/usr/local/Cellar/go/1.15/libexec/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc000721fd0 sp=0xc000721fc8 pc=0x406baa1
created by fyne.io/fyne/internal/driver/glfw.(*gLDriver).startDrawThread
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/internal/driver/glfw/loop.go:182 +0xbd
```

OS: MacOSX Catalina
Fyne: 5065c283

**Comment by sxvghd** [comment]
With this piece of code I can reliably reproduce the crash on my machine.
It seems like it's tied to the number of widgets to keep track of, because it worked perfectly fyne without those 50 buttons inside. 
```
package main

import (
    "fyne.io/fyne/v2"
    "fyne.io/fyne/v2/app"
    "fyne.io/fyne/v2/container"
    "fyne.io/fyne/v2/widget"
    "strconv"
)

func NewWindow(a fyne.App) {
    w := a.NewWindow("I await deaths sweet kiss...")

    hello := widget.NewLabel("Kill me!")
    cont := container.NewVBox()
    w.SetContent(container.NewBorder(
        hello,
        widget.NewButton("Kill!", func() {
            w.Close()
        }),
        nil,
        nil,
        container.NewVScroll(cont,),
    ))
    for i := 0; i < 50; i++ {
        cont.Add(widget.NewButton(strconv.Itoa(i),func(){}))
    }
    w.Show()
}

func main() {
    a := app.New()
    w := a.NewWindow("Hello")

    hello := widget.NewLabel("Hello Fyne!")
    w.SetContent(container.NewVBox(
        hello,
        widget.NewButton("Hi!", func() {
            NewWindow(a)
        }),
    ))

    w.ShowAndRun()
}
```
Additionally, info and log:
OS: Void Linux
Fyne: 2.0
```
>> go run crash.go 
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xf8 pc=0x7f1c4963c0d0]

runtime stack:
runtime.throw(0x7dff28, 0x2a)
	/usr/lib/go/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/usr/lib/go/src/runtime/signal_unix.go:726 +0x4ac

goroutine 24 [syscall, locked to thread]:
runtime.cgocall(0x720d40, 0xc0000abd28, 0xf4a9b040)
	/usr/lib/go/src/runtime/cgocall.go:133 +0x5b fp=0xc0000abcf8 sp=0xc0000abcc0 pc=0x4297fb
fyne.io/fyne/v2/vendor/github.com/go-gl/gl/v3.2-core/gl._Cfunc_glowClear(0x7f1c493f5700, 0x3e4243c500004100)
	_cgo_gotypes.go:3663 +0x45 fp=0xc0000abd28 sp=0xc0000abcf8 pc=0x668de5
fyne.io/fyne/v2/vendor/github.com/go-gl/gl/v3.2-core/gl.Clear(...)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/vendor/github.com/go-gl/gl/v3.2-core/gl/package.go:8667
fyne.io/fyne/v2/internal/painter/gl.(*glPainter).glClearBuffer(0xc000104c30)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/painter/gl/gl_core.go:188 +0xe6 fp=0xc0000abd50 sp=0xc0000abd28 pc=0x68be66
fyne.io/fyne/v2/internal/painter/gl.(*glPainter).Clear(0xc000104c30)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/painter/gl/painter.go:53 +0x2b fp=0xc0000abd68 sp=0xc0000abd50 pc=0x68c6eb
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).paint(0xc000348100, 0x442cc00044a6c000)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/canvas.go:427 +0x8a fp=0xc0000abde0 sp=0xc0000abd68 pc=0x6b48ea
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/loop.go:164 +0x88 fp=0xc0000abe30 sp=0xc0000abde0 pc=0x6c1688
fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0xc00016a380, 0xc0000abe58)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:1181 +0x38 fp=0xc0000abe48 sp=0xc0000abe30 pc=0x6bfdd8
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0xc000122370, 0xc00016a380)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/loop.go:155 +0x65 fp=0xc0000abe80 sp=0xc0000abe48 pc=0x6b6c85
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread.func1(0xc000122370, 0xc0001243c0, 0xc000122410)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/loop.go:211 +0x245 fp=0xc0000abfc8 sp=0xc0000abe80 pc=0x6c19c5
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc0000abfd0 sp=0xc0000abfc8 pc=0x490221
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/loop.go:182 +0xbd

goroutine 1 [select, locked to thread]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0xc000122370)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/loop.go:90 +0x1bf
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0xc000122370)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/driver.go:79 +0x3a
fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0xc00016a1c0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:447 +0x57
main.main()
	/home/sxvghd/Temp/crash.go:43 +0x235

goroutine 19 [sleep]:
time.Sleep(0x22ecb25c00)
	/usr/lib/go/src/runtime/time.go:188 +0xbf
fyne.io/fyne/v2/internal/painter.svgCacheJanitor.func1()
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/painter/svg_cache.go:57 +0x109
sync.(*Once).doSlow(0xd61948, 0xc00011acb0)
	/usr/lib/go/src/sync/once.go:66 +0xec
sync.(*Once).Do(0xd61948, 0xc00011acb0)
	/usr/lib/go/src/sync/once.go:57 +0x45
created by fyne.io/fyne/v2/internal/painter.svgCacheJanitor
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/painter/svg_cache.go:55 +0x9b

goroutine 20 [chan receive]:
fyne.io/fyne/v2/data/binding.processItems()
	/home/sxvghd/.go/src/fyne.io/fyne/v2/data/binding/queue.go:20 +0x45
created by fyne.io/fyne/v2/data/binding.init.0
	/home/sxvghd/.go/src/fyne.io/fyne/v2/data/binding/queue.go:15 +0x35

goroutine 21 [chan receive]:
fyne.io/fyne/v2/app.newAppWithDriver.func1(0xc0001241e0, 0xc0002da230)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/app/app.go:117 +0x3d
created by fyne.io/fyne/v2/app.newAppWithDriver
	/home/sxvghd/.go/src/fyne.io/fyne/v2/app/app.go:115 +0x1e8

goroutine 22 [syscall]:
syscall.Syscall6(0xe8, 0x7, 0xc000337b6c, 0x7, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/syscall/asm_linux_amd64.s:41 +0x5
fyne.io/fyne/v2/vendor/golang.org/x/sys/unix.EpollWait(0x7, 0xc000337b6c, 0x7, 0x7, 0xffffffffffffffff, 0x0, 0x0, 0x0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/vendor/golang.org/x/sys/unix/zsyscall_linux_amd64.go:76 +0x72
fyne.io/fyne/v2/vendor/github.com/fsnotify/fsnotify.(*fdPoller).wait(0xc00012cc00, 0x0, 0x0, 0x0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/vendor/github.com/fsnotify/fsnotify/inotify_poller.go:86 +0x91
fyne.io/fyne/v2/vendor/github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc0001223c0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/vendor/github.com/fsnotify/fsnotify/inotify.go:192 +0x206
created by fyne.io/fyne/v2/vendor/github.com/fsnotify/fsnotify.NewWatcher
	/home/sxvghd/.go/src/fyne.io/fyne/v2/vendor/github.com/fsnotify/fsnotify/inotify.go:59 +0x1a8

goroutine 23 [chan receive]:
fyne.io/fyne/v2/app.watchFile.func1(0xc0001223c0, 0xc000108150, 0x28, 0xc000101bd0, 0xc000101be0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/app/settings_desktop.go:42 +0x5a
created by fyne.io/fyne/v2/app.watchFile
	/home/sxvghd/.go/src/fyne.io/fyne/v2/app/settings_desktop.go:41 +0xec

goroutine 25 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*window).runEventQueue(0xc00016a1c0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:1241 +0xb0
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:1266 +0x10d

goroutine 26 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).setupThemeListener.func1(0xc000124420, 0xc000348000)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/canvas.go:476 +0x47
created by fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).setupThemeListener
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/canvas.go:474 +0xa6

goroutine 31 [chan receive]:
fyne.io/fyne/v2/internal/painter.SvgCacheMonitorTheme.func1(0xc000124480)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/painter/svg_cache.go:109 +0x34
created by fyne.io/fyne/v2/internal/painter.SvgCacheMonitorTheme
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/painter/svg_cache.go:107 +0x9c

goroutine 50 [chan receive]:
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).setupThemeListener.func1(0xc0001246c0, 0xc000348100)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/canvas.go:476 +0x47
created by fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).setupThemeListener
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/driver/glfw/canvas.go:474 +0xa6

goroutine 8 [chan receive]:
fyne.io/fyne/v2/internal/animation.(*Runner).runAnimations.func1(0xc00008e190, 0xc0001059e0)
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/animation/runner.go:47 +0x226
created by fyne.io/fyne/v2/internal/animation.(*Runner).runAnimations
	/home/sxvghd/.go/src/fyne.io/fyne/v2/internal/animation/runner.go:45 +0x56
exit status 2
```

**Comment by andydotxyz** [comment]
I finally have a handle on this :) PR soon

**Comment by andydotxyz** [comment]
Promoting to blocker as this can be caused by closing a window not just when quitting the app

**Comment by andydotxyz** [comment]
Fix on `develop` and `release/v2.0.x` for testing
